Changed namespace rosSubscribe to rosQueuedSubscribe

ksyy opened the pull request:
To avoid a namespace shadowing.

ksyy committed:
Changed namespace rosSubscribe to rosQueuedSubscribe to avoid namespace shadowing.

ksyy committed:
ReadQueue now uses the dynamic_graph time to begin unqueueing the information received.

jmirabel commented on sot_hpp/ros_interface.py:
```diff
@@ -104,7 +105,17 @@ def readQueue(self, req):
+            answer = self.runCommand (cmd)
+            print cmd
+            print answer
+            return GetIntResponse (int(answer.result))
```
I guess from this change that you intent to use it as follows:
```
t = getSoTTime()
readQueue (t + shift)
```
where `shift>= 1` is an integer.

I fear this two calls to ROS services will add an unpredictable delay between them. Either `shift` is set to a high value or it may happen that `t + shift` is not greater than the new SoT time when readQueue is executed and `readQueue` may run into trouble.

Wouldn't it be better to have only one function `readQueue` like the following ?
```python
def readQueue(shift)
  t = getSoTTime()
  self.supervisor.readQueue(t + shift)
```

There should not be any delay issues. Moreover, I would rather say that exposing SoT time is not desirable.

ksyy committed:
Removed exposition of the SoT time to the user
Removed getSoTTime().
Changed readQueue(time) to readQueue(shift) which gets the SoTTime on its own and starts reading the queues at time + shift.

ksyy committed:
Added ROS service: stopReadingQueue
A bit more intuitive than sending readingQueue(-1).

ksyy commented:
As requested, I removed the exposition of the SoT time and changed the function `readQueue` to work with the `shift` syntax you described.

I also provided a better named function to stop the reading of the queues.

jmirabel commented on sot_hpp/ros_interface.py:
```diff
@@ -99,23 +99,29 @@ def clearQueues(self, req):
+            print cmd
+            print answer
+            SoTTime = int(answer.result)
+            cmd = "supervisor.readQueue(" + str(SoTTime + req.data) + ")"
```
Sorry, I think I was not clear. There is still two ROS calls here as `runCommand` uses ROS services. I meant that it should be done in class Supervisor.

ksyy replied:
Oh, I forgot that `runCommand` used ROS services…
[4290b80](https://github.com/jmirabel/sot-hpp/pull/1/commits/4290b807de34cef751d0adc36d92cca9c7761961) moves the functionality to the class Supervisor and removes the extra call to `runCommand`.

jmirabel commented:
> I also provided a better named function to stop the reading of the queues.

:+1:

ksyy committed:
Moved modifications on readQueue from ros_interface to supervisor to avoid a possible delay issue between ros services calls
